File: 0034-find-first-and-last-position-of-element-in-sorted-array/solution.java
class Solution 
{
    public int[] searchRange(int[] nums, int target) 
    {
        int[] arr={-1,-1};
        arr[0]=findstart(nums,target);
        if(arr[0]!=-1)
        {
            arr[1]=findend(nums,target);
        }
        return arr;
    }
    public static int findstart(int[] arr,int target)
    {
        int l=0,r=arr.length-1;
        int index=-1;
        while(l<=r)
        {
            int mid=(l+r)/2;
            if(arr[mid]>=target)
            {
                r=mid-1;
            }
            else{
                l=mid+1;
            }

            if(arr[mid]==target)
            {
                index=mid;
            }
        }
        return index;
    }

    public static int findend(int[] arr,int target)
    {
        int index=-1;
        int l=0,r=arr.length-1;
        while(l<=r)
        {
            int mid=(l+r)/2;
            if(arr[mid]<=target)
            {
                l=mid+1;
            }
            else{
                r=mid-1;
            }
            if(arr[mid]==target)
            {
                index=mid;
            }
        }
        return index;
    }
}

File: 0034-find-first-and-last-position-of-element-in-sorted-array/solution.cpp
class Solution {
public:
    int second(vector<int>& nums,int target){
        int l=0,r=nums.size()-1;
        int ans=-1;
        while(l<=r){
            int mid=(l+r)/2;
            if(nums[mid]==target){
                ans=mid;
            }
            if(nums[mid]<=target){
                l=mid+1;
            }else{
                r=mid-1;
            }
        }
        return ans;
    }
    int first(vector<int>& nums,int target){
        int l=0,r=nums.size()-1;
        int ans=-1;
        while(l<=r){
            int mid=(l+r)/2;
            if(nums[mid]>=target){
                r=mid-1;
            }else{
                l=mid+1;
            }
            if(nums[mid]==target){
                ans=mid;
            }
        }
        return ans;
    }
    vector<int> searchRange(vector<int>& nums, int target) {
        vector<int> arr={-1,-1};
        arr[0]=first(nums,target);
        arr[1]=second(nums,target);
        return arr;
    }
};
